File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms2/FloatUniform.java
package org.saar.lwjgl.opengl.shaders.uniforms2;

import org.lwjgl.opengl.GL20;

public abstract class FloatUniform extends UniformBase implements Uniform {

    @Override
    public final void doLoad() {
        GL20.glUniform1f(getLocation(), getUniformValue());
    }

    public abstract float getUniformValue();
}

File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms/FloatUniform.kt
package org.saar.lwjgl.opengl.shaders.uniforms

import org.lwjgl.opengl.GL20

abstract class FloatUniform : Uniform {

    final override fun load(location: Int) {
        val value: Float = this.value
        GL20.glUniform1f(location, value)
    }

    abstract val value: Float
}